index/scorch/persister.go

```diff
@@ -482,6 +482,12 @@ func prepareBoltSnapshot(snapshot *IndexSnapshot, tx *bolt.Tx, path string,
 		switch seg := segmentSnapshot.segment.(type) {
 		case segment.PersistedSegment:
 			segPath := seg.Path()
+			if backupHelper != nil {
+				segPath, err = backupHelper(segPath, segmentSnapshot.id)
```

Comment:
I may not be on the same page here, don't we need to handle the backup for the memory segments too? as those are anyways getting persisted as a part of the prepareBoltSnapshot call.

Reply:
We do, if you expand the section below, you'll see the type assertion already handles segment.UnpersistedSegment.  And on that path, backup and regular persist are identical (just write segment to the incoming path) so no changes were necessary.

Reply:
We do, if you expand the section below, you'll see the type assertion already handles segment.UnpersistedSegment. And on that path, backup and regular persist are identical (just write segment to the incoming path) so no changes were necessary.